src/object/srv_obj_remote.c

```diff
@@ -0,0 +1,299 @@
+/**
+ * (C) Copyright 2019 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+/**
+ * object server operations
+ *
+ * This file contains the object server remote object api.
+ */
+#define D_LOGFAC	DD_FAC(object)
+
+#include <uuid/uuid.h>
+
+#include <abt.h>
+#include <daos/rpc.h>
+#include <daos_srv/pool.h>
+#include <daos_srv/rebuild.h>
+#include <daos_srv/container.h>
+#include <daos_srv/vos.h>
+#include <daos_srv/bio.h>
+#include <daos_srv/daos_server.h>
+#include <daos_srv/dtx_srv.h>
+#include "obj_rpc.h"
+#include "obj_internal.h"
+
+struct obj_remote_cb_arg {
+	dtx_exec_shard_comp_cb_t	comp_cb;
+	void				*cb_arg;
+	crt_rpc_t			*parent_req;
+	struct dtx_exec_shard_arg	*shard_arg;
+};
+
+static void
+shard_update_req_cb(const struct crt_cb_info *cb_info)
```

Comment:
The functions for 'update' and 'punch' can be merged to avoid dup code.

Reply:
Yes, agree. Actually I plan to unify those RPC format, to unify those update/query/punch rpc format, then these function will be easier to unify. I will do that in that patch.